temba/flows/views.py

```diff
@@ -320,10 +321,15 @@ def get_context_data(self, **kwargs):
             )
             context["folders"] = folders
             context["request_url"] = self.request.path
-            context["actions"] = self.actions
             context["contact_fields"] = ContactField.user_fields.filter(
                 org=self.request.user.get_org(), is_active=True
             ).order_by("pk")
+
+            # make actions unavailable for viewers
```

Comment:
To remove the viewer permissions, let's make it work like the other modules, using the Django Permissions, E.g. https://github.com/greatnonprofits-nfp/rapidpro/blob/cf047afdc25ccf1f9a04db170076a79d540066a4/templates/flowimages/flowimage_list.haml#L70

And removing the permissions `archive` and `download` from the viewer permissions on settings file https://github.com/greatnonprofits-nfp/rapidpro/blob/cf047afdc25ccf1f9a04db170076a79d540066a4/temba/settings_common.py#L834

Reply:
Refactored using django permissions